app-libs/stf/src/best_energy_helpers.rs

```diff
@@ -11,20 +12,32 @@ pub static RESULTS_DIR: &str = "./records/market_results";
 
 pub fn write_orders(timestamp: &str, orders: &[Order]) -> Result<(), StfError> {
 	let orders_path = format!("{}/{}.json", ORDERS_DIR, timestamp);
-	fs::write(&orders_path, serde_json::to_string(&orders).unwrap())
-		.map_err(|e| StfError::Dispatch(format!("Writing orders {:?}. Error: {:?}", orders, e)))
+	match serde_json::to_string(&orders) {
+		Ok(serialized_orders) => match fs::write(&orders_path, serialized_orders) {
+			Ok(_) => Ok(()),
+			Err(e) =>
+				Err(StfError::Dispatch(format!("Writing orders {:?}. Error: {:?}", orders, e))),
+		},
+		Err(e) =>
+			Err(StfError::Dispatch(format!("Serializing orders {:?}. Error: {:?}", orders, e))),
+	}
 }
```

Comment:
Double nesting is hard to read. It is better pracice to keep this sequential:

```rust
	let orders_serialized = serde_json::to_string(&orders)
		.map_err(|e| StfError::Dispatch(format!("Serializing orders {:?}. Error: {:?}", orders, e)))?;

	fs::write(&orders_path, orders_serialized)
		.map_err(|e| StfError::Dispatch(format!("Writing orders {:?}. Error: {:?}", orders, e)))
```

Reply:
done